app/models/manageiq/providers/redhat/infra_manager/vm.rb

```diff
@@ -61,18 +62,27 @@ def collect_disks
     disks = hardware.disks.map { |disk| "#{disk.storage.ems_ref}/disks/#{disk.filename}" }
     vm_disks = []
 
-    ext_management_system.with_provider_connection do |rhevm|
-      disks.each do |disk|
-        begin
-          vm_disks << Ovirt::Disk.find_by_href(rhevm, disk)
-        rescue Ovirt::MissingResourceError
-          nil
+    unless ext_management_system.nil?
+      ext_management_system.with_provider_connection do |rhevm|
```

Comment:
`ext_management_system.try(:with_provider_connection)` will save you having to indent this whole block again